src/evaluate.cpp

```diff
@@ -887,14 +889,9 @@ namespace {
 #ifdef THREECHECK
             if (pos.is_three_check())
             {
-                switch(pos.checks_given(Them))
-                {
-                case CHECKS_NB:
-                case CHECKS_3:
-                case CHECKS_2:  kingDanger += 2 * kingDanger; break;
-                case CHECKS_1:  kingDanger += kingDanger; break;
-                case CHECKS_0:  kingDanger += kingDanger / 2; break;
-                }
+              int i = std::max(0, 2 - pos.checks_given(Them));
```

Comment:
The type `Checks` seems to me to come straight from a joke about a programmer who, to  comply with company policy, defined constants: 
 ```
#define ONE 1
 #define TWO 2
``` 
etc.